ros2/content.md

```diff
@@ -0,0 +1,265 @@
+
+```console
+$ docker-compose up -d
+```
+
+> Notice that a new network named `ros2_demos` has been created, as can be shown further with:
+
+```console
+$ docker network inspect ros2_demos
+```
+
+> We can monitor the logged output of each container, such as the listener node like so:
+
+```console
+$ docker-compose logs listener
+```
+
+> Finally, we can stop and remove all the relevant containers using docker-compose from the same directory:
+
+```console
+$ docker-compose stop
+$ docker-compose rm
+```
+
+> Note: the auto-generated network, `ros2_demos`, will persist until you explicitly remove it using `docker-compose down`.
+
+
+### Securing ROS 2
+
+Lets build upon the example above by adding authenticated encryption to the message transport. This is done by leveraging [Secure DDS](https://www.omg.org/spec/DDS-SECURITY). We'll use the same ROS 2 docker image to bootstrap the PKI, CAs, and Digitally Signed files.
+
+> Create a script at `~/ros2_demos/keystore/bootstrap_keystore.bash` to bootstrap a keystore and add entries for each node:
+
+``` shell
+#!/usr/bin/env bash
+# Bootstrap ROS 2 keystore
+ros2 security create_keystore ./
+ros2 security create_key ./ talker
+ros2 security create_key ./ listener
+```
+
+> Create a enforcement file at `~/ros2_demos/config.env` to configure ROS 2 Security:
+
+
+``` shell
+# Configure ROS 2 Security
+ROS_SECURITY_NODE_DIRECTORY=/keystore
+ROS_SECURITY_STRATEGY=Enforce
+ROS_SECURITY_ENABLE=true
+ROS_DOMAIN_ID=0
+```
+
+> Use a temporary container to run the keystore bootstraping script in the keystore directory:
+
+
+```console
+$ docker run -it --rm \
+    --env-file ./config.env \
+    --volume ./keystore:/keystore:rw \
+    --workdir /keystore \
+    ros2 bash bootstrap_keystore.bash
+```
+
+> Now modify the original `docker-compose.yml` to use the configured environment and respective keystore entries:
+
+```yaml
+version: '3'
+
+services:
+  talker:
+    build: ./Dockerfile
+    environment:
+      - ./config.env
+    volumes:
+      - ./keystore/talker:/keystore:ro
+    command: ros2 run demo_nodes_cpp talker
+
+  listener:
+    build: ./Dockerfile
+    environment:
+      - ./config.env
+    volumes:
+      - ./keystore/listener:/keystore:ro
+    command: ros2 run demo_nodes_py listener
+```
+
+> Now simply startup docker-compose as before:
+
+``` command
+$ docker-compose up
+```
+
+Note: So far this has only added authenticated encryption, i.e. only  participants with public certificates signed by a trusted CA may join the domain. To enable access control within the secure domain, i.e. restrict which and how topics may be used by participants, more such details can be found [here](https://github.com/ros2/sros2/).
+
+# More Resources
+
+[ROS.org](http://www.ros.org/): Main ROS website  
+[Docs](https://docs.ros2.org/): Core Documentation  
```

Comment:
I replaced the wiki that is ROS1 related to avoid misleading traffic to older unrelated content. Are there any other links we'd like to add here?

Reply:
https://index.ros.org/doc/ros2/ might be a better entry point than docs.ros2.org.

It's the new home of the ROS2 wiki which moved into a generated site so changes were easier to manage with a pull-request workflow.

Reply:
Well, docs.ros2.org has all the nicely organized sphinx pages. I feel the index url is still a jumble.
Is the intent that the index will become a ordered nesting of sphinx project docs, like say python's official doc pages?

Reply:
See https://github.com/docker-library/docs/pull/1381/commits/6d09179854033de0cc43bbf426832b327c6575f5